packages/bridging/src/providers/near-intents/NearIntentsBridgeProvider.ts

```diff
@@ -0,0 +1,369 @@
+    if (!receipt) return null
+
+    const tradeEvent = (await getCowTradeEvents(chainId, receipt.logs)).find((event) => event.orderUid === orderUid)
+    if (!tradeEvent) return null
+
+    // Identify the recipient addresses of the buyToken transfer.
+    // Example: to swap CRV → AVAX via CoW + NEAR Intents, the flow is:
+    //   1. Sell CRV for USDC on CoW.
+    //   2. Sell USDC for AVAX on NEAR Intents.
+    // This filter extracts the ERC20 Transfer logs of the buyToken (e.g. USDC)
+    // that exactly match the buyAmount from the CoW trade.
+    const buyTokenTransfersReceivers = receipt.logs
+      .filter(
+        (log) =>
+          log.address.toLowerCase() === tradeEvent.buyToken.toLowerCase() &&
+          log.topics[0] === '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef' &&
+          log.topics[1] !== '0x0000000000000000000000000000000000000000000000000000000000000000' &&
+          log.topics[2] !== '0x0000000000000000000000000000000000000000000000000000000000000000' &&
+          BigInt(log.data) === BigInt(tradeEvent.buyAmount),
+      )
+      .map((log) => `0x${log.topics[2]?.slice(26)}`)
+
+    // Resolve the unique EOA (Externally Owned Account) among the buyToken transfer recipients.
+    // A deposit address on NEAR Intents must be an EOA (no contract code).
+    // We fetch the code for each address and filter out any contracts.
+    // Expect exactly one valid EOA → the NEAR Intents deposit address.
+    const eoaAddresses = (
+      await Promise.all(buyTokenTransfersReceivers.map((address) => adapter.getCode(address)))
+    ).filter((code) => code === '0x')
+    if (eoaAddresses.length !== 1) {
+      throw new Error('Failed to retrieve the deposit address')
+    }
+
+    // Resolve tokens + status in parallel
+    const depositAddress = eoaAddresses[0]!
+    const [tokens, status] = await Promise.all([this.api.getTokens(), this.api.getStatus(depositAddress)])
+
+    // Unpack quote data
+    const qr = status.quoteResponse?.quoteRequest
+    const quote = status.quoteResponse?.quote
+    const timestampMs = Date.parse(status.quoteResponse?.timestamp ?? '')
+    if (!qr || !quote || Number.isNaN(timestampMs)) {
+      throw new Error('Malformed quote response from NEAR Intents')
+    }
+
+    const quoteTimestamp = Math.floor(timestampMs / 1000)
+
+    // Locate origin/destination tokens
+    const inputToken = tokens.find((t) => t.assetId === qr.originAsset)
+    const outputToken = tokens.find((t) => t.assetId === qr.destinationAsset)
+    if (!inputToken || !outputToken) throw new Error('Token not supported')
+
+    // Normalize to local token shape
+    const adaptedInput = adaptToken(inputToken)
+    const adaptedOutput = adaptToken(outputToken)
+    if (!adaptedInput?.chainId || !adaptedOutput?.chainId) {
+      throw new Error('Token not supported')
+    }
+
+    // Build response
+    return {
+      status: {
+        fillTimeInSeconds: quote.timeEstimate,
```

Comment:
Why default to ETH_ADDRESS? Expected for any of them to be undefined by this point?